components/client/src/main/java/com/hotels/styx/client/StyxHttpClient.java

```diff
@@ -158,7 +159,7 @@ private static Origin originFromRequest(LiveHttpRequest request, Boolean isHttps
                 .orElseGet(() -> {
                     checkArgument(request.url().isAbsolute(), "host header is not set for request=%s", request);
                     return request.url().authority().map(Url.Authority::hostAndPort)
-                            .orElseThrow(() -> new IllegalArgumentException("Cannot send request " + request + " as URL is not absolute and no HOST header is present"));
+                            .orElseThrow(() -> new IllegalArgumentException("Cannot send request " + formatRequest(request) + " as URL is not absolute and no HOST header is present"));
```

Comment:
Hi. StyxHttpClient doesn't really need to format the headers. Just the bare `.toString` is enough. 

The exception will eventually be processed by `HttpErrorStatusCauseLogger`. Let's leave the full message formatting there.

Reply:
I slightly disagree here, as if we don't format the request it won't show any headers (so we cannot really verify that there's no HOST header). It doesn't seem so important now, but In case we implemented HTTP/2, the :authority header could convey the same information so it would be good to print all 'displayable' headers.

Reply:
fixed - removed formatting from all places apart from HttpErrorStatusCauseLogger and HttpRequestMessageLogger